rollrus.go

```diff
@@ -80,28 +87,55 @@ func ReportPanic(token, env string) {
 
 // Fire the hook. This is called by Logrus for entries that match the levels
 // returned by Levels(). See below.
-func (r *Hook) Fire(entry *log.Entry) (err error) {
-	e := fmt.Errorf(entry.Message)
+func (r *Hook) Fire(entry *log.Entry) error {
+	cause, trace := extractError(entry.Data)
+	if cause == nil {
+		cause = fmt.Errorf(entry.Message)
+	}
+
 	m := convertFields(entry.Data)
 	if _, exists := m["time"]; !exists {
 		m["time"] = entry.Time.Format(time.RFC3339)
 	}
 
-	switch entry.Level {
-	case log.FatalLevel, log.PanicLevel:
-		_, err = r.Client.Critical(e, m)
-	case log.ErrorLevel:
-		_, err = r.Client.Error(e, m)
-	case log.WarnLevel:
-		_, err = r.Client.Warning(e, m)
-	case log.InfoLevel:
-		_, err = r.Client.Info(entry.Message, m)
-	case log.DebugLevel:
-		_, err = r.Client.Debug(entry.Message, m)
-	default:
-		return fmt.Errorf("Unknown level: %s", entry.Level)
-	}
+	return r.report(entry, cause, m, trace)
+}
 
+func (r *Hook) report(entry *log.Entry, cause error, m map[string]string, trace []uintptr) (err error) {
```

Comment:
I don't see why a `switch` to encode exactly what a simple `if-else` does... But, this is essentially what I had first.